File: Server/Server/WebSocket/Strategy/IStrategy.cs
using WebSocket.Strategy.Enumerations;

namespace WebSocket.Strategy
{
    /// <summary>
    /// Interface définissant une stratégie de traitement des messages WebSocket
    /// </summary>
    public interface IStrategy
    {
        /// <summary>
        /// Exécute la stratégie de traitement pour un message reçu
        /// </summary>
        /// <param name="player">Le client qui a envoyé le message</param>
        /// <param name="data">Les données du message sous forme de tableau de chaînes</param>
        /// <param name="gameType">Le type de partie concernée</param>
        /// <param name="response">La réponse à envoyer au client (modifiée par référence)</param>
        /// <param name="type">Le type de réponse (modifié par référence)</param>
        public void Execute(IClient player, string[] data, GameType gameType, ref string response, ref string type);
    }
}

File: Client/src/app/Model/Strategy/IStrategy.ts
import { Game } from '../Game';

/**
 * Interface pour les stratégies
 */
export interface IStrategy {
    /**
     * Exécute le code en fonction de la stratégie
     * @param data message envoyé par le serveur
     * @param state état de la partie
     * @param idGame identifiant de la partie
     * @param game partie en cours
     */
    execute(data: string[], idGame: {value: string}, game:Game,):void;
}